sympy/solvers/solvers.py

```diff
@@ -1383,12 +1383,14 @@ def _solve(f, *symbols, **flags):
         # is only a linear solution
         f_num, sol = solve_linear(f, symbols=symbols)
         if f_num is S.Zero:
-            return []
+            return [S.Zero]
```

Comment:
The answer to `solve(0, x)` is not `x==0` -- that's the solution to `solve(x, x)`.